rfcs/0102-moderation-team.md

```diff
@@ -0,0 +1,94 @@
+---
+feature: moderation team
+start-date: 2021-08-18
+author: tomberek
+co-authors: blaggacao
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: https://github.com/NixOS/rfcs/pull/98
+---
+
+# Summary
+[summary]: #summary
+
+Establish a team to perform moderation.
+
+# Motivation
+[motivation]: #motivation
+
+There is not currently any official mechanism for moderation action. It's not
+sustainable to have to call on Graham any time there's a spammer or conflict
+that requires moderation, and we'd like to help the community become more
+self-regulating.
+
+(adopted from #98)
+
+# Detailed design
+[design]: #detailed-design
+
+The Moderation Team is a volunteer group that may receive, invite, and evaluate
+applications to the team or alter the composition at any time with approval
+from the NixOS Foundation board or by any individual choosing to resign. The
```

Comment:
I wonder if it is better to make Foundation approval more implicit; i.e. normally the team decides on the changes and that's it, but Foundation reserves the right to alter the team composition if deemed necessary.

Reply:
I think the NixOS Foundation is free to organize "approval" so it suites their needs. For example, in periods of low controversy w.r.t. the team, approval can take place implicitly through absence of objection. Either way, "approval" ensures, that the foundation can exert checks & balances if need be.

Somewhat intentionally implicit: with the right to approve comes the right to revoke.

Hence, I think the current formulation elegantly covers that particular matter without constraining practicality.

Reply:
No, the right to approve doesn't always imply the right to revoke. (Also, the rights to approve and to revoke do not automatically imply the right to make a change of an own invention).

I think vague implications are often suboptimal when they can be easily misread in good faith.

Reply:
I think you are right, those rights (to revoke) can only be construed if their non-execution would constitute _gross violations of duty_ on behalf of the foundation. That might be very rare and extreme cases only.

@tomberek Maybe we can tersely hint in the text at an interpretation of "approval" that affords the foundation to take a very "hands-off" approach, when situation & context reasonably allows it.

Reply:
Using this thread to continue the discussion about the Foundation's involvement:
(@shlevy @lf- @ryantm )

> My perception is that this decision to give the Foundation power ... the named group is 3 respected people, some servers, and paperwork. It is constructed to help the project succeed but it does not control the existing governance.

The motivation was to have some oversight by people the community trusts, hence the language limiting the role to approval of the composition of the team, not day-to-day matters. Of course, the community always would have some measure of oversight (see @ryantm 's comment below) - what would be a good way to include this thought?

> It would be the case without saying it, but maybe it would help (and be enough by itself) to say authority over the team is retained by the RFC process itself.

@shlevy suggested using the RFC process instead. Did you mean an RFC would determine approval in each case, or just a "when needed to veto" situation? I presume the community at large can always use that mechanism when needed, but using it for the normal rotation of people seems abuse of the RFC process and can lead to significant bikeshedding. 

Proposal: What if approval is via the RFC Steering Committee instead? I could see that being reasonable and working out well. My goal here is to have something more solid than "the entire community", but I'm somewhat agnostic to who/what that is as long as it is "people the community trusts".

Reply:
I don't think we should put more responsibility on the Steering Committee.

Reply:
@tomberek I think my main point is, from a starting point of a limited overall mandate, we should trust the team to figure out what process is appropriate, ultimately subject to intervention via the RFC process. If they deem a formal process is needed, they can propose it. If some non-team contributor sees a problem with the moderation team (including but not limited to new team member selection), they can object informally and then formally if need be.

Reply:
Okay, good points. I'm thinking of something along these lines:

The Moderation Team is a volunteer group that may receive, invite, and evaluate applications to the team or alter the composition at any time~with approval from the NixOS Foundation board or by any individual choosing to resign~. The team's composition, contact information, and announcements should be maintained at https://nixos.org/community/teams/moderation.html. The team shall perform moderation activities on behalf of the community, **with oversight via the RFC process**, for the discussions in the official project spaces as well as unofficial spaces that seek and reach such an agreement with the team. The team should utilize the NixOS Foundation mission and the following statement during their duties:

(removed the resignation section because without the "approval" clause I think it's more clear that "volunteer" implies ability to resign)

Reply:
I like that modification!